src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1293,6 +1299,161 @@ private function registerPropertyAccessConfiguration(array $config, ContainerBui
+                ->addArgument($fetcher['auto_ack'])
+                ->setPublic(false)
+            ;
+
+            $fetchers[$name] = $id;
+        }
+
+        foreach ($config['fetchers']['services'] as $name => $fetcher) {
+            if (isset($fetchers[$name])) {
+                throw new \InvalidArgumentException(sprintf('A fetcher named "%s" already exist.', $name));
+            }
+            $id = "worker.message_fecher.service.$name";
+            $container->setAlias($id, $fetcher['service']);
+            $fetchers[$name] = $id;
+        }
+
+        foreach ($config['fetchers']['buffers'] as $name => $fetcher) {
+            if (!isset($fetchers[$fetcher['wrap']])) {
+                throw new \InvalidArgumentException(sprintf('The fetcher named "%s" could not wrap "%s" as it does not exit.', $name, $fetcher['wrap']));
+            }
+            $id = "worker.message_fecher.buffer.$name";
+            $container
+                ->register($id, Worker\MessageFetcher\BufferedMessageFetcher::class)
+                ->addArgument(new Reference($fetchers[$fetcher['wrap']]))
+                ->addArgument(array(
+                    'max_buffering_time' => $fetcher['max_buffering_time'],
+                    'max_messages' => $fetcher['max_messages'],
+                ))
+                ->setPublic(false)
+            ;
+            $fetchers[$name] = $id;
+        }
+
+        $routers = array();
+
+        foreach ($config['routers']['directs'] as $name => $router) {
+            if (isset($routers[$name])) {
+                throw new \InvalidArgumentException(sprintf('A router named "%s" already exist.', $name));
+            }
+            if (!isset($fetchers[$router['fetcher']])) {
+                throw new \InvalidArgumentException(sprintf('The router named "%s" could not use fetcher "%s" as it does not exit.', $name, $router['fetcher']));
+            }
+            $id = "worker.router.direct.$name";
+            $container
+                ->register($id, Worker\Router\DirectRouter::class)
+                ->addArgument(new Reference($fetchers[$router['fetcher']]))
+                ->addArgument(new Reference($router['consumer']))
+                ->setPublic(false)
+            ;
+
+            $routers[$name] = $id;
+        }
+
+        foreach ($config['routers']['round_robins'] as $name => $router) {
+            $wrappedRouters = array();
+            foreach ($router['groups'] as $wrappedRouter) {
+                if (!isset($routers[$wrappedRouter])) {
+                    throw new \InvalidArgumentException(sprintf('The router named "%s" could not use "%s" as it does not exit.', $name, $wrappedRouter));
+                }
+                $wrappedRouters[] = new Reference($routers[$wrappedRouter]);
+            }
+            $id = "worker.router.round_robin.$name";
+
+            $container
+                ->register($id, Worker\Router\RoundRobinRouter::class)
```

Comment:
I did not use full import on workflow and I think it's better.
It prefer this way.